Source/ProgressionSystemRuntime/Public/Data/PSSaveGameData.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#pragma once
+
+#include "Data/PSTypes.h"
+#include "Templates/Tuple.h"
+#include "GameFramework/SaveGame.h"
+#include "PSSaveGameData.generated.h"
+
+
+/**
+ * Defines the standard process for the saving slots names and index 
+ */
+UCLASS()
+class PROGRESSIONSYSTEMRUNTIME_API UPSSaveGameData : public USaveGame
+{
+	GENERATED_BODY()
+
+public:
+	/** Returns the name of the save slot. */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	static const FString& GetSaveSlotName();
+
+	/** Returns the Slot Index of the save slot. */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	static int32 GetSaveSlotIndex() { return 0; }
+
+	/** Returns the ProgressionRow by Index */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	const FPSRowData& GetSavedProgressionRowByIndex(int32 Index) const;
+
+	/** Set element by index */
+	UFUNCTION(BlueprintCallable, Category= "C++")
+	void SetCurrentProgressionRowByIndex(int32 InIndex);
+
+	/** Update the ProgressionRows map */
+	UFUNCTION(BlueprintCallable, Category = "C++")
+	void SetProgressionMap(const TMap<FName, FPSRowData> ProgressionRows);
+
+	/** Get current progression row */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	const FPSRowData& GetCurrentRow() const;
+
+	/** Get current progression row */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	FORCEINLINE FName GetCurrentRowName() const { return CurrentRowName; }
+
+	/** Set the current level by player tag*/
+	void SetRowByTag(FPlayerTag PlayerTag);
+
+	/** Unlock level by Index, used only for the first level */
+	UFUNCTION(BlueprintCallable, Category = "C++")
+	void UnlockLevelByName(FName RowName);
+
+	/** Unlock level by Index, used only for the first level */
+	UFUNCTION(BlueprintCallable, Category = "C++")
+	void SavePoints(EEndGameState EndGameState);
+
+	/** Unlocks the next level*/
+	void NextLevelProgressionRowData();
+
+	/** Returns the endgame reward. */
+	UFUNCTION(BlueprintCallable, Category="C++")
+	int32 GetProgressionReward(EEndGameState EndGameState);
+
+protected:
+	/** The current Index of Saved Progression. */
+	UPROPERTY()
```

Comment:
1. Do you save this property? If it's runtime cached value, please add 'Transient' specifier, so this property will not be saved.
2. `Internal` postfix
3. `VisibleInstanceOnly, BlueprintReadWrite` for blueprint developers inheriting from this save class

Reply:
> Do you save this property? If it's runtime cached value, please add 'Transient' specifier, so this property will not be saved.

Yes, I do save it. All other changes added

Reply:
For #3 I mean we still need all the rest specifiers like Category (since it's visible in blueprints), BlueprintProtected (to be allowed only in children) and Display Name (to get rid off `Internal` in blueprints)

Reply:
@h4rdmol ping

Reply:
done